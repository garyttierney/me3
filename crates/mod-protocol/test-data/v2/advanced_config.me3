profileVersion = "v2"

[game]
launch = "nightreign"
savefile = "NRMOD.sl2"
disable_arxan = true
start_online = true

[mods]
my_mod.path = './my-mod'
my_dll = { path = './my-mod/my_dll.dll', initializer.delay.ms = 3000 }
hks_debug = { path = './hks_debug.me3', optional = true }
my_other_mod = { path = './my-other-mod', disabled = true }
my_profile = { path = 'my_profile.me3', optional = false, disabled = true }
my_other_dll.path = 'other_dll.dll'
my_other_dll.initializer.function = "init_my_dll"
