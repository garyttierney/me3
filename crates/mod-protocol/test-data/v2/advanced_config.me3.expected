V2(
    ModProfileV2 {
        supports: Some(
            Nightreign,
        ),
        natives: [
            Native {
                inner: ModFile {
                    name: "my_dll",
                    path: "./my-mod/my_dll.dll",
                    enabled: true,
                    optional: false,
                },
                initializer: Some(
                    NativeInitializerCondition {
                        delay: Some(
                            NativeInitializerDelay {
                                ms: 3000,
                            },
                        ),
                        function: None,
                    },
                ),
                load_before: [],
                load_after: [],
            },
            Native {
                inner: ModFile {
                    name: "my_other_dll",
                    path: "other_dll.dll",
                    enabled: true,
                    optional: false,
                },
                initializer: Some(
                    NativeInitializerCondition {
                        delay: None,
                        function: Some(
                            "init_my_dll",
                        ),
                    },
                ),
                load_before: [],
                load_after: [],
            },
        ],
        packages: [
            Package {
                inner: ModFile {
                    name: "my_mod",
                    path: "./my-mod",
                    enabled: true,
                    optional: false,
                },
                load_before: [],
                load_after: [],
            },
            Package {
                inner: ModFile {
                    name: "my_other_mod",
                    path: "./my-other-mod",
                    enabled: true,
                    optional: false,
                },
                load_before: [],
                load_after: [],
            },
        ],
        profiles: [
            ModFile {
                name: "hks_debug",
                path: "./hks_debug.me3",
                enabled: true,
                optional: true,
            },
            ModFile {
                name: "my_profile",
                path: "my_profile.me3",
                enabled: true,
                optional: false,
            },
        ],
        savefile: Some(
            "NRMOD.sl2",
        ),
        start_online: Some(
            true,
        ),
        disable_arxan: Some(
            true,
        ),
    },
)
